### PR TITLE
Move Dynamite over to the new UpdateExpression API

### PR DESCRIPTION
**Pull request by nicks**
Hello @xiao, @adrianlee44, 

Please review the following commits I made in branch 'nicks/expr'.

7993582f2f0b59637018859277ee2ecd9853e417 (2017-01-17 17:24:46 -0500)
Move Dynamite over to the new UpdateExpression API

R=@xiao
R=@adrianlee44

### CODE REVIEW

File: lib/DynamoRequest.js [hunk 1]
```diff
@@ -270,6 +274,10 @@ DynamoRequest.prototype.build = function () {
     delete this.data.FilterExpression
   }
 
+  if (!this.data.ConditionExpression) {
```
Comment (adrianlee44) [inline lib/DynamoRequest.js]: Delete the key if doesn't have any value?

File: lib/UpdateExpressionBuilder.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
+var typeUtil = require('./typeUtil')
+
+// The UpdateExpression API has different names for all the action types, but they're
+// semantically the same as the old actionsl
```
Comment (nicks) [inline lib/UpdateExpressionBuilder.js]: 😬

File: lib/UpdateExpressionBuilder.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+function getActionName(attr) {
+  if (attr.Action == 'PUT') {
+    return 'SET'
+  } else if (attr.Action == 'ADD') {
+    return 'ADD'
+  } else if (attr.Action == 'DELETE') {
+    return attr.Value ? 'DELETE' : 'REMOVE'
+  } else {
+    throw new Error('Unrecognized action ' + attr.Action)
+  }
```
Comment (nicks) [inline lib/UpdateExpressionBuilder.js]: ya i like that better

File: lib/DynamoRequest.js (hunk 1 above)
Comment (nicks) [inline lib/DynamoRequest.js]: yep. oh and i need to add UpdateExpression too

**Pull request merged**